### PR TITLE
fix(openai): mark openai.request span as errored when AI Guard blocks

### PR DESCRIPTION
**Pull request by avara1986**
### What does this PR do?

When AI Guard blocks an OpenAI call, the `AIGuardAbortError` was raised on the child `ai_guard` span, but the parent `openai.request` (LLM Obs) span stayed green. This PR propagates the block to the LLM span so it is correctly marked as errored.

The After Model evaluation now runs **before** `finish()`, so a block rejects into the existing `.catch`, which finishes the span with the `AIGuardAbortError`. This also keeps `openai.request` wrapped around its child `ai_guard` span (it previously closed before the After Model evaluation completed) and binds the lazy `APIPromise` `parse`/`asResponse` to the LLM span async context so Before/After Model `ai_guard` spans nest under `openai.request`.

### Additional Notes

- Added `assertLlmSpanErrored()` to the AI Guard integration tests and asserted `openai.request` is errored on both Before Model and After Model OpenAI deny paths.
- Verified locally: AI Guard integration suite passes (4 deny incl. both After Model cases + 12 allow/edge-case), lint clean.

Before, two separate spans:
image
image

After, with this fix:
image

### CODE REVIEW

File: packages/datadog-instrumentations/src/helpers/openai-ai-guard.js
Comment (IlyasShabi) [inline packages/datadog-instrumentations/src/helpers/openai-ai-guard.js:1]: IMO this solution is not ideal because it relies on `AsyncResource.bind()` to preserve the right active span across lazy OpenAI promise methods. That makes it depend on async context behavior at a delayed execution point, which is easier to break if the SDK call path changes.

A better approach is to pass the parentSpan through to the AI Guard SDK and use it as an explicit childOf when creating the ai guard span, for example:
```
  const traceOpts = childOf ? { childOf } : {}
  return this.#tracer.trace(TAGS.RESOURCE, traceOpts, async (span) => {

```

That way the parent-child relationship is set directly when the AI Guard span is created.

Comment (IlyasShabi) [reply packages/datadog-instrumentations/src/helpers/openai-ai-guard.js:1]: Here is a draft for potential simpler/safe solution https://github.com/DataDog/dd-trace-js/commit/cb81a0a50261e40f8458bd197bac8db7b9edad3a

File: packages/datadog-instrumentations/src/openai.js [hunk 1]
```diff
@@ -240,6 +240,11 @@ for (const extension of extensions) {
           }
 
           return ch.start.runStores(ctx, () => {
+            // Capture the active `openai.request` span (set by the plugin's `ch.start`
+            // subscriber) so AI Guard nests `ai_guard` spans under it via explicit `childOf`,
+            // without relying on async-context binding at evaluation time.
```
Comment (CarlesDD) [inline packages/datadog-instrumentations/src/openai.js]: Nit: comment explains what the code does rather than why

```suggestion
            // Explicit childOf rather than async-context: the _thenUnwrap/parse path
            // decouples the lazy evaluation from the active scope at call time.
```

File: packages/datadog-instrumentations/src/openai.js [hunk 2]
```diff
@@ -300,23 +305,33 @@ function handleUnwrappedAPIPromise (apiProm, ctx, stream, guard) {
...
+      if (!guard) {
+        finish(ctx, responseData)
+        return body
+      }
 
-      return aiGuard.evaluateOutput(guard, body).then(() => body)
+      // Run After Model AI Guard before finishing the span. When it blocks it rejects, so the
+      // outer catch finishes the span with the AIGuardAbortError and marks openai.request (the
+      // LLM Obs span) as errored. Finishing here also keeps openai.request wrapped around its
+      // child `ai_guard` span instead of closing before the After Model evaluation completes.
```
Comment (CarlesDD) [inline packages/datadog-instrumentations/src/openai.js]: Nit: comment explains what the code does rather than why

```suggestion
      // Finish after evaluation so a block propagates the error to openai.request
      // and the span wraps its ai_guard child instead of closing before it.
```

**Review by CarlesDD — DISMISSED** [review]
LGTM. The `finish()` after `evaluateOutput` ordering is the right fix; the span duration increase is an expected trade-off. I've added two comment style nits.

**Review by CarlesDD — APPROVED** [review]

**Review by IlyasShabi — APPROVED** [review]
